src/logic/render/PolygonRenderEngine.ts

```diff
@@ -15,36 +40,325 @@ export class PolygonRenderEngine extends BaseRenderEngine {
+                        return indexUnderMouse;
+                    }, null);
 
-    public mouseMoveHandler(data: EditorData): void {}
+                    if (anchorIndex !== null) {
+                        this.startExistingLabelResize(data, polygonUnderMouse.id, anchorIndex);
+                    } else {
+                        this.updateActivelyCreatedLabel(data);
+                    }
+                } else {
+                    this.updateActivelyCreatedLabel(data);
+                }
+            }
+        }
+    }
+
+    public mouseUpHandler(data: EditorData): void {
+        if (this.isResizeInProgress())
+            this.endExistingLabelResize(data);
+    }
+
+    public mouseMoveHandler(data: EditorData): void {
+        if (!!data.activeImageRectOnCanvas && !!data.mousePositionOnCanvas) {
+            const isOverImage: boolean = RectUtil.isPointInside(data.activeImageRectOnCanvas, data.mousePositionOnCanvas);
+            if (isOverImage && !this.isCreationInProgress()) {
+                const labelPolygon: LabelPolygon = this.getPolygonUnderMouse(data);
+                if (!!labelPolygon) {
+                    if (store.getState().editor.highlightedLabelId !== labelPolygon.id) {
+                        store.dispatch(updateHighlightedLabelId(labelPolygon.id))
+                    }
+                } else {
+                    if (store.getState().editor.highlightedLabelId !== null) {
+                        store.dispatch(updateHighlightedLabelId(null))
+                    }
+                }
+            }
+        }
+    }
 
     // =================================================================================================================
     // RENDERING
     // =================================================================================================================
 
     public render(data: EditorData): void {
-        this.updateCursorStyle(data);
+        const imageData: ImageData = EditorSelector.getActiveImageData();
+        if (imageData) {
+            this.drawExistingLabels(data);
+            this.drawActivelyCreatedLabel(data);
+            this.drawActivelyResizedLabel(data);
+            this.updateCursorStyle(data);
+        }
     }
 
     private updateCursorStyle(data: EditorData) {
         if (!!this.canvas && !!data.mousePositionOnCanvas) {
-            if (RectUtil.isPointInside(data.activeImageRectOnCanvas, data.mousePositionOnCanvas)) {
-                store.dispatch(updateCustomcursorStyle(CustomCursorStyle.DEFAULT));
+            if (RectUtil.isPointInside({x: 0, y: 0, ...CanvasUtil.getSize(this.canvas)}, data.mousePositionOnCanvas)) {
+                if (this.isCreationInProgress()) {
+                    const isMouseOverStartAnchor: boolean = this.isMouseOverAnchor(data.mousePositionOnCanvas, this.activePath[0]);
+                    if (isMouseOverStartAnchor)
+                        store.dispatch(updateCustomcursorStyle(CustomCursorStyle.MOVE));
```

Comment:
i jak wdrożysz (oczywiście nie w ramach tego zadania) te akcje @dispatchable które Ci podesłałem to to będzie pięknie wyglądać wtedy